datafuselabs/databend
# [metasrv] temporarily disable connection pool to avoid connection hang

I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

Temporarily disable connection pool to avoid connection hang. As [tonic doc said](https://docs.rs/tower-service/0.3.1/tower_service/trait.Service.html#tymethod.poll_ready) if `poll_ready` return `Poll::Ready(Ok(()))` that service is able to process requests. But if metasrv is force killed, that TCP connection remains half-open and the `poll_ready` still return `Poll::Ready(Ok(()))`.

### how to reproduce
0. add some debug info to [check function](https://github.com/datafuselabs/databend/blob/7385f416059bc7d5bc4eaa9c77c5d01cf659c197/common/meta/grpc/src/grpc_client.rs#L114)
1. deploy databend-query and databend-meta manually
2. using MySQL cli connect to databend and do some create table and insert oprations.
3. kill databend-meta and perform some operation with MySQL cli, then you can find it hang on...
4. using `strace` on the hanged process see it stuck on [`recvfrom`](https://linux.die.net/man/3/recvfrom)

## Changelog

- Improvement

## Related Issues

Fixes #issue
